app/src/androidTest/java/com/android/sample/ui/listActivities/ListActivitiesTest.kt

```diff
@@ -289,6 +315,53 @@ class OverviewScreenTest {
     composeTestRule.onNodeWithTag("likeButtontrue").assertIsDisplayed()
   }
 
+  @Test
+  fun distanceIsCorrectlyDisplayedInMeters() {
+    userProfileViewModel = mock(ProfileViewModel::class.java)
+    composeTestRule.setContent {
+      ActivityCard(
+          navigationActions = navigationActions,
+          listActivitiesViewModel = listActivitiesViewModel,
+          profileViewModel = userProfileViewModel,
+          profile = testUser,
+          activity = activity,
+          distance = 0.5503f)
+    }
+    composeTestRule
+        .onNodeWithTag("distanceText", useUnmergedTree = true)
+        .assertTextContains("Distance : 550m")
+  }
+
```

Comment:
So Great that you test the UI now ! Plus you also test the rounding which is great.